### PR TITLE
Private chats

### PR DESCRIPTION
**Pull request by Rafer45**
Begin work on private chats

I added a conversation set for each user. For now, a user's conversation set only contains any conversations directly created by the user. The set is serialized and stored in persistent storage.

### CODE REVIEW

File: src/main/java/codeu/controller/ConversationServlet.java [hunk 1]
```diff
@@ -117,6 +117,10 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
         new Conversation(UUID.randomUUID(), user.getId(), conversationTitle, Instant.now());
 
     conversationStore.addConversation(conversation);
+
+    // Users are always whitelisted in conversations they create
+    user.addToConversation(conversation.getId());
```
Comment (eselenic) [inline src/main/java/codeu/controller/ConversationServlet.java:122]: Great catch!

File: src/main/java/codeu/model/data/User.java [hunk 2]
```diff
@@ -56,6 +60,8 @@ public User(UUID id, String name, String passwordHash, Instant creation, boolean
     this.passwordHash = passwordHash;
     this.creation = creation;
     this.isAdmin = isAdmin;
+
+    this.conversationSet = new HashSet<>();
```
Comment (eselenic) [inline src/main/java/codeu/model/data/User.java:64]: Add a parameter for conversationSet

**Review by eselenic — CHANGES_REQUESTED** [review]

File: src/main/java/codeu/model/data/User.java (hunk 2 above)
Comment (Rafer45) [reply src/main/java/codeu/model/data/User.java:64]: I initially did that, but then realized I would have to refactor every single user creation instance.

A conversation set can be set with `setConversationSet` (starting to feel like Dr. Seuss with all this assonance)

Comment (eselenic) [reply src/main/java/codeu/model/data/User.java:64]: Good point!

**Comment by eselenic** [comment]
LGTM!

**Review by eselenic — APPROVED** [review]

**Pull request merged**